tests/js_test.go

```diff
@@ -303,6 +303,95 @@ func TestInternalizeDate(t *testing.T) {
 	}
 }
 
+func TestInternalizeStruct(t *testing.T) {
+	type Person struct {
+		Name string
+		Age  int
+	}
+	var a Person
+
+	js.Global.Set("return_person", func(p *Person) *Person {
+		if p == nil {
+			t.Fail() // Or t.Error("Received nil pointer")
+			return nil
+		}
+		a = *p
+		return p
+	})
+
+	js.Global.Call("eval", "return_person({Name: 'foo', Age: 952})")
+
+	if a.Name != "foo" || a.Age != 952 {
+		t.Fail()
+	}
```

Comment:
Here and below: please use https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff to do the comparison.

Reply:
done